### ISSUE TITLE
allow status page jobs to run concurrently

### ISSUE THREAD

**Issue by tfoote** [ros-infrastructure/ros_buildfarm#331]
On first inspection they look concurrent safe, but testing them overnight for #327 exposed two different backtraces once they were running in parallel. 

```
# BEGIN SECTION: Run Dockerfile - status page
05:48:55 + rm -fr /home/jenkins-slave/workspace/Irel_release-status-page/debian_repo_cache
05:48:55 + rm -fr /home/jenkins-slave/workspace/Irel_release-status-page/status_page
05:48:55 + mkdir -p /home/jenkins-slave/workspace/Irel_release-status-page/debian_repo_cache
05:48:55 + mkdir -p /home/jenkins-slave/workspace/Irel_release-status-page/status_page
05:48:55 + docker run --rm --cidfile=/home/jenkins-slave/workspace/Irel_release-status-page/docker_generate_status_page/docker.cid --net=host -v /home/jenkins-slave/workspace/Irel_release-status-page/ros_buildfarm:/tmp/ros_buildfarm:ro -v /home/jenkins-slave/workspace/Irel_release-status-page/debian_repo_cache:/tmp/debian_repo_cache -v /home/jenkins-slave/workspace/Irel_release-status-page/status_page:/tmp/status_page status_page_generation
05:48:56 The build file contains the following targets:
05:48:56   - xenial source
05:48:56   - xenial armhf
05:48:59 Traceback (most recent call last):
05:48:59   File "/tmp/ros_buildfarm/scripts/status/build_release_status_page.py", line 48, in <module>
05:48:59     main()
05:48:59   File "/tmp/ros_buildfarm/scripts/status/build_release_status_page.py", line 44, in main
05:48:59     args.cache_dir, args.output_dir, copy_resources=args.copy_resources)
05:48:59   File "/tmp/ros_buildfarm/ros_buildfarm/status_page.py", line 76, in build_release_status_page
05:48:59     building_repo_url, targets, cache_dir)
05:48:59   File "/tmp/ros_buildfarm/ros_buildfarm/debian_repo.py", line 31, in get_debian_repo_data
05:48:59     debian_repository_baseurl, target, cache_dir)
05:48:59   File "/tmp/ros_buildfarm/ros_buildfarm/debian_repo.py", line 47, in get_debian_repo_index
05:48:59     fetch_gzip_url(url, cache_filename)
05:48:59   File "/tmp/ros_buildfarm/ros_buildfarm/debian_repo.py", line 79, in fetch_gzip_url
05:48:59     with open(dst_filename, 'wb') as f:
05:48:59 PermissionError: [Errno 13] Permission denied: '/tmp/debian_repo_cache/5b337d33138df528adc8e61d2ea1067a'
```
```
# BEGIN SECTION: Run Dockerfile - status page
05:33:53 + rm -fr /var/lib/jenkins/jobs/Krel_release-status-page/workspace/debian_repo_cache
05:33:53 + rm -fr /var/lib/jenkins/jobs/Krel_release-status-page/workspace/status_page
05:33:53 + mkdir -p /var/lib/jenkins/jobs/Krel_release-status-page/workspace/debian_repo_cache
05:33:53 + mkdir -p /var/lib/jenkins/jobs/Krel_release-status-page/workspace/status_page
05:33:53 + docker run --rm --cidfile=/var/lib/jenkins/jobs/Krel_release-status-page/workspace/docker_generate_status_page/docker.cid --net=host -v /var/lib/jenkins/jobs/Krel_release-status-page/workspace/ros_buildfarm:/tmp/ros_buildfarm:ro -v /var/lib/jenkins/jobs/Krel_release-status-page/workspace/debian_repo_cache:/tmp/debian_repo_cache -v /var/lib/jenkins/jobs/Krel_release-status-page/workspace/status_page:/tmp/status_page status_page_generation
05:33:54 python3: can't open file '/tmp/ros_buildfarm/scripts/status/build_release_status_page.py': [Errno 13] Permission denied
```

Some example failed jobs both kinetic, on the new label, and indigo on the old label, on the slave_on_master executor still. 
- https://groups.google.com/forum/#!topic/ros-buildfarm-indigo/LNh3Q52GKO4
- https://groups.google.com/forum/#!topic/ros-buildfarm-indigo/iF_UD2y6XQw
- https://groups.google.com/forum/#!topic/ros-buildfarm-indigo/NAp_Jb-59ao
- https://groups.google.com/forum/#!topic/ros-buildfarm-kinetic/PZO-wYTbbkQ
- https://groups.google.com/forum/#!topic/ros-buildfarm-kinetic/EqnAQmnVO5E

**Comment by dirk-thomas** [comment]
I don't see a need to run these jobs in parallel atm. They only run once every 15 min. and finish way within this time frame.

If you are not planning to look into this enhancement and create a pull request I would close this ticket as *wontfix*.

**Comment by tfoote** [comment]
This will significantly improve my experience to not be blocked by long running tasks. As mentioned https://github.com/ros-infrastructure/ros_buildfarm/pull/327#issuecomment-241042499 it could be unblocked by making a new executor with a different label but that's increasing complexity of the deployment when this does not have fundamental concurrency problem and will make our lives easier in general if it's execution does not require coordination.

**Comment by dirk-thomas** [comment]
I will close this ticket for now since I don't think it is feasible to run these jobs in parallel. They are running on the same slave as the sync jobs in order to not in parallel to a sync. Otherwise the status page will show pretty broken information since the apt state is in flux.
